cmd/frontend/graphqlbackend/user.go

```diff
@@ -190,6 +192,114 @@ func (r *UserResolver) CodyProEnabled(ctx context.Context) bool {
 	return r.CodyProEnabledAt(ctx) != nil
 }
 
+func (r *UserResolver) CodyCurrentPeriodChatUsage(ctx context.Context) (*int32, error) {
+	if !envvar.SourcegraphDotComMode() {
+		return nil, errors.New("this feature is only available on sourcegraph.com")
+	}
+
+	currentPeriodStartDate, err := r.CodyCurrentPeriodStartDate()
+	if err != nil {
+		return nil, err
+	}
+
+	query := sqlf.Sprintf(`WHERE user_id = %s AND timestamp >= %s AND timestamp <= NOW() AND (name LIKE '%%recipe%%' OR name LIKE '%%command%%' OR name LIKE '%%chat%%')`, r.user.ID, currentPeriodStartDate.Time)
+
+	count, err := r.db.EventLogs().CountBySQL(ctx, query)
+
+	intCount := int32(count)
+
+	return &intCount, err
+}
+
+func (r *UserResolver) CodyCurrentPeriodCodeUsage(ctx context.Context) (*int32, error) {
+	if !envvar.SourcegraphDotComMode() {
+		return nil, errors.New("this feature is only available on sourcegraph.com")
+	}
+
+	currentPeriodStartDate, err := r.CodyCurrentPeriodStartDate()
+	if err != nil {
+		return nil, err
+	}
+
+	query := sqlf.Sprintf(`WHERE user_id = %s AND timestamp >= %s AND timestamp <= NOW() AND name LIKE '%%completion:suggested%%'`, r.user.ID, currentPeriodStartDate.Time)
+
+	count, err := r.db.EventLogs().CountBySQL(ctx, query)
+
+	intCount := int32(count)
+
+	return &intCount, err
+}
+
+func (r *UserResolver) CodyCurrentPeriodStartDate() (*gqlutil.DateTime, error) {
+	if !envvar.SourcegraphDotComMode() {
+		return nil, errors.New("this feature is only available on sourcegraph.com")
+	}
+
+	// 🚨 SECURITY: Only the user and admins are allowed to access the user's
+	// settings, because they may contain secrets or other sensitive data.
```

Comment:
Nit to satisfy my spellchecker 😄 
```suggestion
	// settings because they may contain secrets or other sensitive data.
```